<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Обзор-данных" data-toc-modified-id="Обзор-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Обзор данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Обработка-пропусков" data-toc-modified-id="Обработка-пропусков-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Обработка пропусков</a></span></li><li><span><a href="#Обработка-аномальных-значений" data-toc-modified-id="Обработка-аномальных-значений-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Обработка аномальных значений</a></span></li><li><span><a href="#Обработка-пропусков-(продолжение)" data-toc-modified-id="Обработка-пропусков-(продолжение)-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Обработка пропусков (продолжение)</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-2.6"><span class="toc-item-num">2.6&nbsp;&nbsp;</span>Категоризация данных</a></span></li></ul></li><li><span><a href="#Исследовательский-анализ-данных" data-toc-modified-id="Исследовательский-анализ-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Исследовательский анализ данных</a></span><ul class="toc-item"><li><span><a href="#Влияние-количества-детей-на-возврат-кредита-в-срок" data-toc-modified-id="Влияние-количества-детей-на-возврат-кредита-в-срок-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Влияние количества детей на возврат кредита в срок</a></span></li><li><span><a href="#Влияние-семейного-положения-на-возврат-кредита-в-срок" data-toc-modified-id="Влияние-семейного-положения-на-возврат-кредита-в-срок-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Влияние семейного положения на возврат кредита в срок</a></span></li><li><span><a href="#Влияние-уровня-дохода-на-возврат-кредита-в-срок" data-toc-modified-id="Влияние-уровня-дохода-на-возврат-кредита-в-срок-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Влияние уровня дохода на возврат кредита в срок</a></span></li><li><span><a href="#Влияние-целей-кредита-на-возврат-кредита-в-срок" data-toc-modified-id="Влияние-целей-кредита-на-возврат-кредита-в-срок-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Влияние целей кредита на возврат кредита в срок</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

# Исследование надежности заемщиков

**Заказчик — кредитный отдел банка.** 

**Цель исследования — установить влияние различных факторов на факт погашения кредита в срок.** 

**Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.** 

**Исходные данные от банка — статистика о платёжеспособности клиентов.** 

**Описание данных:**
- `children` — количество детей в семье
- `days_employed` — общий трудовой стаж в днях
- `dob_years` — возраст клиента в годах
- `education` — уровень образования клиента
- `education_id` — идентификатор уровня образования
- `family_status` — семейное положение
- `family_status_id` — идентификатор семейного положения
- `gender` — пол клиента
- `income_type` — тип занятости
- `debt` — имел ли задолженность по возврату кредитов
- `total_income` — ежемесячный доход
- `purpose` — цель получения кредита

*Импортируем необходимые для работы библиотеки.*

In [1]:
import pandas as pd

## Обзор данных

*Прочитаем csv-файл с исходными данными и запишем его в переменную `data`.*

In [2]:
data = pd.read_csv('data.csv')

*Выводим первые 20 строк датафрейма на экран.*

In [4]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

*Выводим основную информацию о датафрейме.*

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


**Выводы:**

1. Каждая строка датафрейма содержит различные сведения о конкретном клиенте банка.
2. В столбцах `days_employed` и `total_income` имеются пропуски.

## Предобработка данных

### Обработка пропусков

*Выведем количество пропущенных значений для каждого столбца.*

In [6]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

В столбце `total_income` хранятся данные о доходах. На сумму дохода сильнее всего влияет тип занятости,
поэтому заполнить пропуски в этом столбце можно медианным значением по каждому типу из столбца `income_type`.
Например, у человека с типом занятости сотрудник пропуск в столбце `total_income` должен быть заполнен
медианным доходом среди всех записей с тем же типом.

*Проходим циклом по уникальным значениям столбца `income_type`, считаем медиану по столбцу `total_income`, для каждого уникального значения столбца `income_type` заменяем `Nan` на медианные значения.*

In [7]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. Проверим данные на аномалии и обработаем их.

В столбце `days_employed` находятся данные о количестве дней трудового стажа. \
**Среди них встречаются аномалии - отрицательные значения.** \
*Заменим все отрицательные значения положительными с помощью метода `abs()`:*

In [8]:
data['days_employed'] = data['days_employed'].abs()

*Для каждого типа занятости выведем медианное значение трудового стажа `days_employed` в днях.*

In [9]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

**У двух типов (безработные и пенсионеры) получатся аномально большие значения.** \
Исправить такие значения сложно, поэтому оставим их как есть. Тем более этот столбец не понадобится вам для исследования.

*Выведем перечень уникальных значений столбца `children`.*

In [10]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

**В столбце `children` есть два аномальных значения (-1 и 20).** \
*Удалим строки, в которых встречаются такие аномальные значения из датафрейма `data`.*

In [11]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

*Выведем перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.*

In [12]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Обработка пропусков (продолжение)

*Заполним пропуски в столбце `days_employed` медианными значениями по каждому типу занятости `income_type`.*

In [13]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

*Убедимся, что все пропуски заполнены. Выведем количество пропущенных значений для каждого столбца.*

In [14]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

*Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.*

In [15]:
data['total_income'] = data['total_income'].astype(int)

In [17]:
data['total_income'].dtype

dtype('int32')

### Обработка дубликатов

*Выведем на экран количество строк-дубликатов в данных.*

In [18]:
data.duplicated().sum()

54

*Удалим строки с дубликатами.*

In [19]:
data = data.drop_duplicates()

*Обработаем неявные дубликаты в столбце `education`.
В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв.
Приведем их к нижнему регистру.*

In [20]:
data['education'] = data['education'].str.lower()

### Категоризация данных

*Cоздадим в датафрейме `data` новый столбец `total_income_category` с категориями:*

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


*Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используем собственную функцию с именем `categorize_income()` и метод `apply()`.*

In [21]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [22]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

*Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose`.*

In [23]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

*Cоздадим в датафрейме `data` на основании данных из столбца `purpose` новый столбец `purpose_category`,
в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

*Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.*

*Используем собственную функцию с именем `categorize_purpose()` и метод `apply()`.*

In [64]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [65]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

*Выведем таблицу с данными на экран.*

In [26]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              среднее             1   
8          2    6929.865299         35               высшее             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   
12         0  365286.622650         65              среднее             1   
13         0    1846.641941         54  неоконченное высшее             2   
14         0    1844.956182         56               высшее             0   
15         1     972.364419         26              среднее             1   
16         0    1719.934226         35              среднее             1   
17         0    2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0   10038.818549         48              среднее             1   

       family_status  family_status_id gender income_type  debt  total_income  \
0    женат / замужем                 0      F   сотрудник     0        253875   
1    женат / замужем                 0      F   сотрудник     0        112080   
2    женат / замужем                 0      M   сотрудник     0        145885   
3    женат / замужем                 0      M   сотрудник     0        267628   
4   гражданский брак                 1      F   пенсионер     0        158616   
5   гражданский брак                 1      M   компаньон     0        255763   
6    женат / замужем                 0      F   компаньон     0        240525   
7    женат / замужем                 0      M   сотрудник     0        135823   
8   гражданский брак                 1      F   сотрудник     0         95856   
9    женат / замужем                 0      M   сотрудник     0        144425   
10   женат / замужем                 0      M   компаньон     0        113943   
11   женат / замужем                 0      F   сотрудник     0         77069   
12  гражданский брак                 1      M   пенсионер     0        118514   
13   женат / замужем                 0      F   сотрудник     0        130458   
14  гражданский брак                 1      F   компаньон     1        165127   
15   женат / замужем                 0      F   сотрудник     0        116820   
16   женат / замужем                 0      F   сотрудник     0        289202   
17  гражданский брак                 1      M   сотрудник     0         90410   
18    вдовец / вдова                 2      F   пенсионер     0         56823   
19         в разводе                 3      F   сотрудник     0        242831   

                                   purpose total_income_category  \
0                            покупка жилья                     B   
1                  приобретение автомобиля                     C   
2                            покупка жилья                     C   
3               дополнительное образование                     B   
4                          сыграть свадьбу                     C   
5                            покупка жилья                     B   
6                        операции с жильем                     B   
7                              образование                     C   
8                    на проведение свадьбы                     C   
9   

**Выводы:**

1. Пропуски в столбцах `total_income` и `days_employed` заполнены медианными значениями по каждому типу занятости из столбца `income_type`.
2. Обработаны аномальные значения в столбцах `days_employed` и `children`.
3. Вещественный тип данных в столбце `total_income` изменён на целочисленный.
4. Удалены явные дубликаты.
5. Обработаны неявные дубликаты в столбце `education`.
6. Добавлен столбец `total_income_category` с категориями клиентов по величине дохода.
7. Добавлен столбец `purpose_category` с категориями клиентов по цели кредита.

## Исследовательский анализ данных

### Влияние количества детей на возврат кредита в срок

*Посчитаем общее количество заёмщиков, количество и долю должников от общего количества заёмщиков для каждой категории по количеству детей.*

In [53]:
borrowers_with_children = data.groupby('children') \
                              .agg(all_clients=('children', 'count'),
                                   clients_with_depts=('debt', 'sum')) \
                              .reset_index()

borrowers_with_children['debtors_share'] = pd.Series(['{:.1%}'.format(i) for i in \
                                                      borrowers_with_children['clients_with_depts'] / \
                                                      borrowers_with_children['all_clients']])

*Отсортируем заёмщиков каждой категории по доле должников и выведем результат.*

In [57]:
borrowers_with_children.sort_values(by='debtors_share', 
                                    ascending=False)

children  all_clients  clients_with_depts debtors_share
4         4           41                   4          9.8%
2         2         2052                 194          9.5%
1         1         4809                 444          9.2%
3         3          330                  27          8.2%
0         0        14107                1063          7.5%
5         5            9                   0          0.0%

**Вывод: установлено, что наибольшая доля должников (9.8%) наблюдается в категории заёмщиков с четырьмя детьми. Однако, их доля незначительно отличается от доли должников в категориях с одним ребёнком (9.2%) и с двумя детьми (9.5%). Среди заёмщиков с пятью детьми должников не оказалось, что связано, по-видимому, с небольшой длиной выборки (9 заёмщиков). Наименьшая доля должников (7.5%) наблюдается среди заёмщиков без детей. В рамках данного исследования можно сделать вывод, что заемщики без детей чаще возвращают кредит в срок.**

### Влияние семейного положения на возврат кредита в срок

*Посчитаем количество заёмщиков, количество и долю должников от общего количества заёмщиков для каждой категории по семейному положению.*

In [51]:
borrowers_family_status = data.groupby('family_status') \
                              .agg(all_clients=('family_status', 'count'),
                                   clients_with_depts=('debt', 'sum')) \
                              .reset_index()

borrowers_family_status['debtors_share'] = pd.Series(['{:.1%}'.format(i) for i in \
                                                      borrowers_family_status['clients_with_depts'] / \
                                                      borrowers_family_status['all_clients']])

*Отсортируем заёмщиков каждой категории по доле должников и выведем результат.*

In [58]:
borrowers_family_status.sort_values(by='debtors_share', 
                                    ascending=False)

family_status  all_clients  clients_with_depts debtors_share
0  Не женат / не замужем         2796                 273          9.8%
3       гражданский брак         4146                 385          9.3%
4        женат / замужем        12266                 927          7.6%
1              в разводе         1189                  84          7.1%
2         вдовец / вдова          951                  63          6.6%

**Вывод: наибольшая доля должников наблюдается среди неженатых (9.8%) и состоящих в гражданском браке (9.3%) заёмщиков. Наименьшая доля должников наблюдается среди вдовцов (6.6%). Среди женатых заемщиков должников 7.6%, среди разведённых 7.1%. В рамках данного исследования можно сделать вывод, что вдовцы чаще возвращают кредит в срок, затем по платежеспособности
следуют женатые (7.6%) и состоящие в разводе (7.1%), чаще долги по кредитам чаще бывают состоящих в гражданском браке
(9.3%) и неженатых заёмщиков.**

### Влияние уровня дохода на возврат кредита в срок

*Посчитаем кколичество заёмщиков, количество и долю должников от общего количества заёмщиков для каждой категории по уровню доходов.*

In [61]:
borrowers_total_income = data.groupby('total_income_category') \
                             .agg(all_clients=('total_income_category', 'count'),
                                  clients_with_depts=('debt', 'sum')) \
                             .reset_index()

borrowers_total_income['debtors_share'] = pd.Series(['{:.1%}'.format(i) for i in \
                                                     borrowers_total_income['clients_with_depts'] / \
                                                     borrowers_total_income['all_clients']])

*Отсортируем заёмщиков каждой категории по доле должников и выведем результат.*

In [62]:
borrowers_total_income.sort_values(by='debtors_share', 
                                   ascending=False)

total_income_category  all_clients  clients_with_depts debtors_share
4                     E           22                   2          9.1%
2                     C        15938                1353          8.5%
0                     A           25                   2          8.0%
1                     B         5014                 354          7.1%
3                     D          349                  21          6.0%

**Вывод: наименьшая доля должников (6%) наблюдается среди заёмщиков с доходами от 30001 до 50000, наибольшая (9.1 %) среди 
заёмщиков с доходами до 30000. Среди заемщиков с доходами свыше 1000000 доля должников составляет 8.0%, 
что немногим меньше доли должников среди заемщиков с доходом от 50001 до 200000 (8.5%). 
Доля должников среди заёмщиков с доходами от 200001 до 1000000 несколько меньше и составляет 7.1%. 
В рамках данного исследования можно сделать вывод, что у заемщиков с самыми низкими доходами 
чаще бывают долги по кредитам, однако заемщики с немногим более высокими доходами (от 30001 до 50000) чаще выплачивают 
кредиты в срок.**

### Влияние целей кредита на возврат кредита в срок

*Посчитаем количество заёмщиков, количество и долю должников от общего количества заёмщиков для каждой категории по цели кредита.*

In [66]:
borrowers_purpose_category = data.groupby('purpose_category') \
                                 .agg(all_clients=('purpose_category', 'count'),
                                      clients_with_depts=('debt', 'sum')) \
                                 .reset_index()

borrowers_purpose_category['debtors_share'] = pd.Series(['{:.1%}'.format(i) for i in \
                                                         borrowers_purpose_category['clients_with_depts'] / \
                                                         borrowers_purpose_category['all_clients']])

*Отсортируем заёмщиков каждой категории по доле должников и выведем результат.*

In [67]:
borrowers_purpose_category.sort_values(by='debtors_share', 
                                       ascending=False)

purpose_category  all_clients  clients_with_depts debtors_share
0    операции с автомобилем         4281                 400          9.3%
2     получение образования         3989                 369          9.3%
3        проведение свадьбы         2324                 183          7.9%
1  операции с недвижимостью        10754                 780          7.3%

**Выводы: наиболее часто задолженности по кредитам бывают у заемщиков, которые берут кредиты на операции с автомобилем 
и получение образования (по 9.3%). Реже задолженности по кредитам возникают у заемщиков, которые берут кредиты на операции 
с недвижимостью (7.3%) и на проведение свадьбы (7.9%).**

## Общий вывод

**Целью исследования было установление влияния различных факторов на платежеспособность заемщика.**

**В рамках данного исследования можно сделать следующие выводы:**

1. Наличие любого количества детей влияет на платежеспособность скорее отрицательно. Чаще возвращающают кредит в срок заемщики, не имеющие детей (7.5% от общего числа заемщиков без детей). У заемщиков, имеющих детей, долги по кредитам случаются несколько чаще (8.5-9.8%) от общего числа заемщиков сдетьми).
2. Семейное положение влияет на платежеспособность так: что вдовцы чаще всех (6.6% от общего числат заемщиков-вдовцов) возвращают кредит в срок, долги по кредитам чаще бывают у неженатых заёмщиков (9.8% от общего числа неженатых заемщиков).
3. Уровень доходов влияет на платежеспособность следующим образом: наименьшая доля должников (6% от общего числа заемщиков в этой категории доходов) наблюдается среди заёмщиков с доходами от 30001 до 50000, наибольшая (9.1% от общеготчисла заемщиков в этой категории доходов) среди заёмщиков с доходами до 30000. 
4. Влияние цели кредита на платежеспособность таково: наиболее часто задолженности по кредитам бывают у заемщиков, которые берут кредиты на операции с автомобилем и получение образования (по 9.3% от общего числа заемщиков в каждой категории). Реже задолженности по кредитам возникают у заемщиков, которые берут кредиты на операции с недвижимостью (7.3% от общего числа заемщиков в категории).
